### DISCUSSION TITLE
How can flip webcam image

### DISCUSSION THREAD

**Discussion by DavidMGomez** [PeculiarVentures/GammaCV#198] [Q&A]
Hello I hope you are well!, could someone help me solve a question, how can I flip the image I get from the webcam using the library API,

**Comment by WorldThirteen** [comment]
Hi @DavidMGomez!

GammaCV does not have a built-in simple flip operation, but there are still a few ways you can achieve this:
- build your own flip operation;
- use perspectiveProjection with specifically configured matrix to achieve flip functionality, here is a quick example I've gathered for you:
```js
// configure flip parameters
const flipX = true;
const flipY = false;
  
// generate projection matrix acording to flip setting
const tMatrix = new gm.Tensor('float32', [3, 1, 4], new Float32Array([
  flipX ? -1 : 1, 0, flipX ? width : 0, 0,
  0, flipY ? -1 : 1, flipY ? height : 0, 0,
  0, 0, 1, 0,
]));
// use the image as the input for the matrix transform operation
const operation = gm.perspectiveProjection(input, tMatrix, [height, width, 4]);
```
Link for a demo: https://codepen.io/WorldThirteen/pen/OJGZwZq
